pkg/resources/management.cattle.io/v3/project/validator.go

```diff
@@ -94,20 +100,34 @@ func (a *admitter) admitDelete(project *v3.Project) (*admissionv1.AdmissionRespo
 	return admission.ResponseAllowed(), nil
 }
 
-func (a *admitter) admitCreateOrUpdate(oldProject, newProject *v3.Project) (*admissionv1.AdmissionResponse, error) {
-	fieldErr, err := a.checkClusterName(newProject)
+func (a *admitter) admitCreate(project *v3.Project) (*admissionv1.AdmissionResponse, error) {
+	fieldErr, err := a.checkClusterExists(project)
 	if err != nil {
 		return nil, fmt.Errorf("error checking cluster name: %w", err)
 	}
 	if fieldErr != nil {
 		return admission.ResponseBadRequest(fieldErr.Error()), nil
 	}
+	return a.admitCommonCreateUpdate(nil, project)
+}
+
+func (a *admitter) admitUpdate(oldProject, newProject *v3.Project) (*admissionv1.AdmissionResponse, error) {
+	if oldProject.Spec.ClusterName != newProject.Spec.ClusterName {
+		reason := fmt.Sprintf("clusterName is immutable, was: %s, is now: %s", oldProject.Spec.ClusterName, newProject.Spec.ClusterName)
```

Comment:
Nit: update reason to match Upstream and CRTB validation
https://github.com/rancher/webhook/blob/release/v0.4/pkg/resources/management.cattle.io/v3/clusterroletemplatebinding/validator.go#L133C1-L133C1
```
The Job "update-job" is invalid: spec.template: Invalid value: 
core.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", 
UID:"", ResourceVersion:"", 
.... 
: field is immutable
```
```suggestion
		reason := "field is immutable"
```